moped-editor/src/auth/user.js

```diff
@@ -114,32 +114,44 @@ export const initializeUserDBObject = async (session) => {
   const sessionDataFromLocalStorage = getSessionDatabaseData();
 
   // If the session is valid and there is no existing data...
-  if (session && sessionDataFromLocalStorage === null) {
+  if (sessionDataFromLocalStorage === null) {
     // Fetch the data from Hasura
-    fetch(config.env.APP_HASURA_ENDPOINT, {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-        Authorization: `Bearer ${token}`,
-        "X-Hasura-Role": `${getHighestRole(session)}`,
-      },
-      body: JSON.stringify({
-        query: ACCOUNT_USER_PROFILE_GET_PLAIN,
-        variables: {
-          userId: getDatabaseId(session),
+    try {
+      const res = await fetch(config.env.APP_HASURA_ENDPOINT, {
```

Comment:
i ended up converting this to async/await since I'm finding it easier to follow then the promise chains.